apps/desktop-wallet/src/features/sentTransactions/sentTransactionsSlice.ts

```diff
@@ -0,0 +1,49 @@
+/*
+Copyright 2018 - 2024 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { createSlice, EntityState } from '@reduxjs/toolkit'
+
+import { sentTransactionStatusChanged } from '@/features/sentTransactions/sentTransactionsActions'
+import { sentTransactionsAdapter } from '@/features/sentTransactions/sentTransactionsAdapter'
+import { receiveTestnetTokens } from '@/storage/global/globalActions'
+import { transactionSent } from '@/storage/transactions/transactionsActions'
+import { activeWalletDeleted, walletLocked, walletSwitched } from '@/storage/wallets/walletActions'
+import { SentTransaction } from '@/types/transactions'
+
+type SentTransactionsState = EntityState<SentTransaction>
+
+const initialState: SentTransactionsState = sentTransactionsAdapter.getInitialState()
+
+const sentTransactionsSlice = createSlice({
+  name: 'sentTransactions',
+  initialState,
+  reducers: {},
+  extraReducers(builder) {
+    builder
+      .addCase(transactionSent, sentTransactionsAdapter.addOne)
+      .addCase(receiveTestnetTokens.fulfilled, sentTransactionsAdapter.addOne)
+      .addCase(sentTransactionStatusChanged, (state, { payload: { hash, status } }) => {
+        sentTransactionsAdapter.updateOne(state, { id: hash, changes: { status } })
+      })
+      .addCase(walletLocked, () => initialState)
+      .addCase(walletSwitched, () => initialState)
+      .addCase(activeWalletDeleted, () => initialState)
```

Comment:
Nit: we could use `builder.addMatcher(isAnyOf...` to DRY things off.

Reply:
Done!